### DISCUSSION TITLE
About the nature of Transfers

### DISCUSSION THREAD

**Discussion by floli** [firefly-iii/firefly-iii#8849] [Firefly III]
Hello,

I am importing my statements from two accounts, A and B, using Nordigen/GoCardless with the external identifier as duplicate detector.

Bookings that transfer A -> B are correctly detected as transfers, but created twice, once in A, once in B. That is known problem and addressed in https://docs.firefly-iii.org/how-to/data-importer/import/transfers/.

> By default, Firefly III will skip saving the second transfer because the first one already exists. The second is recognized as a duplicate because all the fields are the same. This may not always be the case. Examples that will stop this from happening are:
> 
> The second transfer has another internal transaction reference (bunq does this).
> [...]

Obviously GoCardless does this as well (or is it my bank?).

> You can however, create custom rules that trigger on any content in the second transfer, and then delete it.

I assume such a rule looks like: `if type == "transfer" and source_account =="A" and description == "something unique" then delete`. The last clause to not make Firefly delete all transfers from A to B.

**Now the questions:** :smile:  When I look at "All Transfers" listing, only one transfer is show, whereas in the list of bookings of either A or B two transfers are shown for each account. The external id of the one listed in transfers matches one of the bookings.

- Why is only one transfer booking shown? Is there some other duplicate transfer detection that takes place?
- Data model wise, does a transfer belongs to A or B, or both? Thus, does it make a difference from which account I delete a duplicate transfer booking?

Thanks for reading this long posting and thanks for your replies!

**Comment by JC5** [comment]
Heya, thanks for your questions!

> When I look at "All Transfers" listing, only one transfer is show, whereas in the list of bookings of either A or B two transfers are shown for each account.

- This means that the two transfers are not actually transfers but withdrawals or deposits.
- It belongs to both, so deleting can be done from either, it won't make a difference
